differ/differ.go

```diff
@@ -294,6 +294,86 @@ func evaluateFilterExpression(eventFilter string, thresholds EventThresholds, ev
 	return evaluator.Evaluate(eventFilter, values)
 }
 
+func produceEntriesToKafka(cluster types.ClusterEntry, ruleContent types.RulesMap,
+	reportItems []types.ReportItem, storage Storage, report types.ClusterReport) int {
+
+	notificationMsg := generateInstantNotificationMessage(
+		&notificationClusterDetailsURI,
+		fmt.Sprint(cluster.AccountNumber),
+		fmt.Sprint(cluster.OrgID),
+		string(cluster.ClusterName))
+	notifiedAt := types.Timestamp(time.Now())
+
+	for _, r := range reportItems {
+		module := r.Module
+		ruleName := moduleToRuleName(module)
+		errorKey := r.ErrorKey
+		likelihood, impact, totalRisk, description := findRuleByNameAndErrorKey(ruleContent, ruleName, errorKey)
+		eventValue := EventValue{
+			Likelihood: likelihood,
+			Impact:     impact,
+			TotalRisk:  totalRisk,
+		}
+
+		// try to evaluate event filter expression
+		result, err := evaluateFilterExpression(kafkaEventFilter,
+			kafkaEventThresholds, eventValue)
+
+		if err != nil {
+			log.Err(err).Msg(evaluationErrorMessage)
+			continue
+		}
+
+		if result > 0 {
+			log.Warn().
+				Str("type", r.Type).
+				Str("rule", string(ruleName)).
+				Str("error key", string(errorKey)).
+				Int("likelihood", likelihood).
+				Int("impact", impact).
+				Int(totalRiskAttribute, totalRisk).
+				Msg("Report with high impact detected")
+			if !shouldNotify(cluster, r, types.NotificationBackendTarget) {
+				continue
+			}
+			// if new report differs from the older one -> send notification
+			log.Info().Str(clusterName, string(cluster.ClusterName)).Msg("Different report from the last one")
+			ReportWithHighImpact.Inc()
+			notificationPayloadURL := generateNotificationPayloadURL(&notificationRuleDetailsURI, string(cluster.ClusterName), module, errorKey)
+			appendEventToNotificationMessage(notificationPayloadURL, &notificationMsg, description, totalRisk, time.Time(cluster.UpdatedAt).UTC().Format(time.RFC3339Nano))
+		}
+	}
+
+	if len(notificationMsg.Events) == 0 {
+		updateNotificationRecordSameState(storage, cluster, report, notifiedAt, types.NotificationBackendTarget)
+		return 0
+	}
+
+	log.Info().Msgf("Producing instant notification for cluster %s with %d events", string(cluster.ClusterName), len(notificationMsg.Events))
+
+	msgBytes, err := json.Marshal(notificationMsg)
+	if err != nil {
+		log.Error().Err(err).Msg(invalidJSONContent)
+		return 0
+	}
+	_, offset, err := kafkaNotifier.ProduceMessage(msgBytes)
+	if err != nil {
+		log.Error().
+			Str(errorStr, err.Error()).
+			Msg("Couldn't send notification message to kafka topic.")
+		updateNotificationRecordErrorState(storage, err, cluster, report, notifiedAt, types.NotificationBackendTarget)
+		os.Exit(ExitStatusKafkaProducerError)
```

Comment:
I think we shouldn't use `os.Exit` here. Returning an error should be enough

Reply:
@juandspy interesting, we commented about it before, right? (AFAIK)

Reply:
I think it was a different place in code, but yes, you commented on it before. Sorry, will change it.

Reply:
> LGTM in general. I think that using the producer and event target as a parameter would make this function more flexible. This way we won't need to rewrite it for the service log producer

Unfortunatelly as it is, the Service Log message needs a little bit more things to be done - e.g. rendering the messages. There is still a lot of common code, though.